mountpoint-s3-fs/src/logging.rs

```diff
@@ -100,14 +122,11 @@ fn install_panic_hook() {
     }))
 }
 
-fn init_tracing_subscriber(config: LoggingConfig) -> anyhow::Result<()> {
-    // Create the logging config from the MOUNTPOINT_LOG environment variable or the default config
-    // if that variable is unset.
-    let env_filter =
-        EnvFilter::try_from_env("MOUNTPOINT_LOG").unwrap_or_else(|_| EnvFilter::new(&config.default_filter));
+fn init_tracing_subscriber(config: LoggingConfig) -> anyhow::Result<LoggingHandle> {
     // Don't create the files or subscribers if we'll never emit any logs
-    if env_filter.max_level_hint() == Some(LevelFilter::OFF) {
-        return Ok(());
+    let default_filter = make_default_filter(config.default_filter.clone())();
+    if default_filter.max_level_hint() == Some(LevelFilter::OFF) {
```

Comment:
We can disable this optimization and set up the listeners all the time maybe to support going from no log to some logging?

Reply:
Is there a performance impact?

Reply:
I expect it to be minimal if there is even. I can try doing simple benchmarks to see if it increases memory usage or decreases throughput. The upside would be going from no log to some level of logging in case something goes wrong with a particular instance, which I feel like it could be useful.